src/components/Cluster/NewCluster/ReleaseSelector/ReleaseSelector.tsx

```diff
@@ -34,46 +28,83 @@ import {
 
 import ReleaseRow from './ReleaseRow';
 
-interface IReleaseSelector {
-  selectRelease(releaseVersion: string): void;
+function getLatestReleaseVersion(
+  releaseVersions: string[],
+  releases: Record<string, IRelease>
+): string | null {
+  if (releaseVersions.length < 1) return null;
 
-  selectedRelease: string;
-  collapsible?: boolean;
-  autoSelectLatest?: boolean;
-  versionFilter?: (version: string) => boolean;
+  for (const version of releaseVersions) {
+    if (releases[version]?.active && !isPreRelease(version)) {
+      return version;
+    }
+  }
+
+  return releaseVersions[0];
 }
 
 const K8sReleaseComponentLabel = styled(ReleaseComponentLabel)`
   margin-left: ${({ theme }) => theme.spacingPx}px;
   margin-bottom: 0;
 `;
 
-const ReleaseSelector: FC<IReleaseSelector> = ({
+export interface IReleaseComponent {
+  name: string;
+  version: string;
+}
+
+export interface IRelease {
+  components: IReleaseComponent[];
+  version: string;
+  active: boolean;
+  timestamp: string;
+  kubernetesVersion?: string;
+  releaseNotesURL?: string;
+  k8sVersionEOLDate?: string;
+}
+
+interface IReleaseSelectorProps {
+  selectRelease: (releaseVersion: string) => void;
+  releases: Record<string, IRelease>;
```

Comment:
This component now takes in the releases from props, so we can use it for both implementations (MAPI and GS API)